### DISCUSSION TITLE
why my access is still blocked by isp?

### DISCUSSION THREAD

**Discussion by furynerd** [abhinavsingh/proxy.py#1430] [Q&A]
hi, i am in a LAN env which has limited access to the internet. luckily i can access to a cloud server，where i run the proxy with 

```bash
python -m proxy --hostname 0.0.0.0 --port 8899
```

when i use this proxy on cloud server to try to access some block site by the LAN, i got msg like:

```bash
...
2024-06-25 09:00:54,754 - pid:11670 [I] tcp_server.handle_readables:189 - Connection reset by client xxx.xxx.xxx.xxx:ppp1
2024-06-25 09:00:54,754 - pid:11670 [I] tcp_server.handle_readables:189 - Connection reset by client xxx.xxx.xxx.xxx:ppp2
2024-06-25 09:00:54,754 - pid:11670 [I] tcp_server.handle_readables:189 - Connection reset by client xxx.xxx.xxx.xxx:ppp3
...
```
where `xxx.xxx.xxx.xxx` looks like some IP from my ISP. and `ppp` is a serial of different port numbers.

but with this proxy,  i can normally acess the sites just already accessible in LAN without proxy.

this make me confused. i suppose all my data traffic should goto proxy on server first, but why still blocked by the ISP?
is a solution? thanks very much

**Comment by abhinavsingh** [comment]
Try `DoH` and see if that helps https://github.com/abhinavsingh/proxy.py?tab=readme-ov-file#cloudflarednsresolverplugin

**Reply by furynerd** [reply]
thanks very much for the hint, but i tried with no improvement...
i installed as suggested:
```bash
pip install "httpx[http2]"
```

then
```bash
python -m proxy --hostname 0.0.0.0 --port 8899 --plugins proxy.plugin.CloudflareDnsResolverPlugin
```

also tried 

```bash
python -m proxy --hostname 0.0.0.0 --port 8899 --plugins proxy.plugin.CustomDnsResolverPlugi
```

but all the results are the same as before

**Reply by abhinavsingh** [reply]
That's unfortunate.   I am usually able to bypass ISP DNS resolver by using the DoH plugin.   I also make sure my router is also pointing to DoH servers hosted by Cloudflare.   You might want to try proxy pool facility, so that all your traffic gets resolved outside of your host system and see if that helps your cause.
